### PR TITLE
[o2k] all Kong entities have should tags

### PR DESCRIPTION
**Pull request by dimitropoulos**
closes: INS-659

### CODE REVIEW

File: packages/openapi-2-kong/src/declarative-config/services.test.ts [hunk 1]
```diff
@@ -14,170 +14,167 @@ const getSpecResult = (): DCService =>
...
-        spec.paths['/cats'].post[xKongPluginRequestValidator] = {
-          config: {
-            parameter_schema: 'operation',
-          },
-        };
-      }
+      (spec.paths['/cats'].post as OA3Operation)[xKongPluginRequestValidator] = {
+        name: 'request-validator',
+        config: {
+          // @ts-expect-error TODO
```
Comment (develohpanda) [inline packages/openapi-2-kong/src/declarative-config/services.test.ts]: What needs to be done here?

File: packages/openapi-2-kong/src/declarative-config/tags.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    services.forEach(service => {
+      service.routes.forEach(route => {
+        expect(route.tags).toEqual(tags);
+      });
+    });
+  });
+
+  it('test that tags are appended to Plugin entities', () => {
+    const spec = getSpec();
+    const keyAuthPlugin: KeyAuthPlugin = {
```
Comment (develohpanda) [inline packages/openapi-2-kong/src/declarative-config/tags.test.ts:29]: Should this test add tags to the plugin itself as well, and make sure the result is a combination of tags in the plugin and tags sent to `generatePlugins`?

File: packages/openapi-2-kong/src/types/openapi3.ts [hunk 3]
```diff
@@ -170,29 +117,29 @@ export type OA3Operation = {
   deprecated?: boolean;
   security?: OA3SecurityRequirement[];
   servers?: OA3Server[];
-} & XKongName
+} & Taggable
```
Comment (develohpanda) [inline packages/openapi-2-kong/src/types/openapi3.ts:127]: What is the rationalle for this being taggable using this interface? It's not taggable in the sense of Kong tags as we use them. There's a distinction between OpenAPI tags and Kong tags and I think this needs to represent it

File: packages/openapi-2-kong/src/types/openapi3.ts [hunk 4]
```diff
@@ -274,15 +221,13 @@ export type OpenApi3Spec = {
   servers?: OA3Server[];
   components?: OA3Components;
   security?: OA3SecurityRequirement[];
-  tags?: string[];
   externalDocs?: OA3ExternalDocs;
-}
+} & Taggable
```
Comment (develohpanda) [inline packages/openapi-2-kong/src/types/openapi3.ts]: Same as my comment above regarding tags on `OA3Operation`

File: packages/openapi-2-kong/README.md [hunk 5]
```diff
@@ -515,36 +559,30 @@ by using custom extensions. The custom extensions are
...
-    - extended by: `x-kong-security-basic-auth`
-    - properties set from OpenAPI spec:
-        - none
-    - requires to add credentials to Kong, which is not supported through
-      OpenAPI specs.
+  - only `Basic` scheme is supported
+  - implemented using the Kong plugin `basic-auth`
+  - extended by: `x-kong-security-basic-auth`
+  - properties set from OpenAPI spec:
+    - none
```
Comment (develohpanda) [inline packages/openapi-2-kong/README.md:584]: I presume a lot of the changes in this file were automated indenting. How did you do that, and is there a way we can automate it?

**Review by develohpanda — APPROVED** [review]
Looks good overall, a couple of questions. The main one is how the OpenAPI spec and OA3Operation types are typed with `Taggable` but this isn't blocking

File: packages/openapi-2-kong/src/declarative-config/services.test.ts (hunk 1 above)
Comment (dimitropoulos) [reply packages/openapi-2-kong/src/declarative-config/services.test.ts]: I tried to switch it to body_schema instead, and it didn't work, so these were still TODO.  I've updated in 52a8697c2a0a699f9246e8b0410fec3294dc38fe

File: packages/openapi-2-kong/src/declarative-config/tags.test.ts (hunk 2 above)
Comment (dimitropoulos) [reply packages/openapi-2-kong/src/declarative-config/tags.test.ts:29]: good call: 77ad3641c46b3fa78ae3965a64210c65f0dc27f3

File: packages/openapi-2-kong/README.md (hunk 5 above)
Comment (dimitropoulos) [reply packages/openapi-2-kong/README.md:584]: yes and yes.  I have markdownlint.  it's definitely in the back of my mind to automate this, but I'd rather spend my energy getting through the typescript conversion first (to phase 4).  I usually trash other changes when it does this, but I didn't bother this time.

File: packages/openapi-2-kong/src/types/openapi3.ts (hunk 4 above)
Comment (dimitropoulos) [reply packages/openapi-2-kong/src/types/openapi3.ts]: This one is just plain wrong.  I updated it in a2e676613e292f1bff095b1e04feeb384424ba84

File: packages/openapi-2-kong/README.md (hunk 5 above)
Comment (develohpanda) [reply packages/openapi-2-kong/README.md:584]: No biggie, linear ticket. Doesn't need to happen right now, but I've never had fun with formatting markdown and this seemed to be automated

File: packages/openapi-2-kong/src/types/openapi3.ts (hunk 3 above)
Comment (dimitropoulos) [reply packages/openapi-2-kong/src/types/openapi3.ts:127]: the rationale is that the spec says it's taggable: https://swagger.io/specification/#operation-object

see: 3eb114cb4c8b2d1703e4f8fe5835e135d2a9f4b7

**Pull request merged**